### PR TITLE
ARS-160 Added the WhoAreYouAgent page and a flag to hide trader radio…

### PR DESCRIPTION
**Pull request by mar36tin**
… button

### Type of change
- [ ] Breaking change
- [x] Non-breaking change
- [ ] Cosmetic change

### Description
Jira link: [Jira title](https://jira.tools.tax.service.gov.uk/browse/ARSSTB-160)

Added the new WhoAreYouAgent page and also introduced the feature flag to hide the radio option for trader

### Attachments or Screenshots
image

### CODE REVIEW

File: app/controllers/AdaptMethodController.scala [hunk 1]
```diff
@@ -51,10 +51,7 @@ class AdaptMethodController @Inject() (
   def onPageLoad(mode: Mode, draftId: DraftId): Action[AnyContent] =
     (identify andThen getData(draftId) andThen requireData) {
       implicit request =>
-        val preparedForm = request.userAnswers.get(AdaptMethodPage) match {
-          case None        => form
-          case Some(value) => form.fill(value)
-        }
+        val preparedForm = AdaptMethodPage.fill(form)
```
Comment (RawToast) [inline app/controllers/AdaptMethodController.scala:54]: This should disappear on a rebase 👍

File: app/forms/RemoveSupportingDocumentFormProvider.scala [hunk 2]
```diff
@@ -19,18 +19,13 @@ package forms
...
-import play.api.data.Forms.set
 
 import forms.mappings.Mappings
-import models.RequiredInformation
 
-class RequiredInformationFormProvider @Inject() extends Mappings {
+class RemoveSupportingDocumentFormProvider @Inject() extends Mappings {
 
-  def apply(): Form[Set[RequiredInformation]] =
+  def apply(): Form[Boolean] =
```
Comment (RawToast) [inline app/forms/RemoveSupportingDocumentFormProvider.scala:27]: I believe this is also something that will go away on a rebase too. This was probably not changed in this PR and just reflects changes on main (in a strange way)

File: app/controllers/WhoAreYouAgentController.scala [hunk 3]
```diff
@@ -46,18 +46,20 @@ class WhoAreYouAgentController @Inject() (
...
+  val form           = formProvider()
+  private val logger = play.api.Logger(getClass)
 
   def onPageLoad(mode: Mode, draftId: DraftId): Action[AnyContent] =
     (identify andThen getData(draftId) andThen requireData) {
       implicit request =>
-        val _ = request.userAnswers.get(WhoAreYouAgentPage) match {
+        logger.info("WhoAreYouAgentController onPageLoad")
+
+        val preparedForm = request.userAnswers.get(WhoAreYouAgentPage) match {
```
Comment (RawToast) [inline app/controllers/WhoAreYouAgentController.scala:57]: I don't think the scaffolding uses this, but there's a handy function for this: `WhoAreYouAgentPage.fill(form)`

File: app/models/WhoAreYouAgent.scala [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+ * limitations under the License.
+ */
+
+package models
+
+/*
+ * Copyright 2023 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
```
Comment (RawToast) [inline app/models/WhoAreYouAgent.scala:23]: Something weird has happened to the copyright notice 😖

File: test/navigation/NavigatorSpec.scala [hunk 5]
```diff
@@ -95,17 +95,26 @@ class NavigatorSpec extends SpecBase {
...
+        ) mustBe routes.WhoAreYouAgentController
+          .onPageLoad(NormalMode, draftId)
+      }
+
+      "should navigate to WhatIsYourRole page for an Agent" in {
+        navigator.nextPage(
+          AccountHomePage,
+          NormalMode,
+          userAnswersAsIndividualTrader.setFuture(AccountHomePage, Agent).futureValue
+//        ) mustBe routes.WhatIsYourRoleAsImporterController.onPageLoad(NormalMode, draftId)
```
Comment (RawToast) [inline test/navigation/NavigatorSpec.scala:116]: I guess this can be removed?

File: app/navigation/Navigator.scala [hunk 6]
```diff
@@ -84,21 +84,30 @@ class Navigator @Inject() () {
...
+        WhoAreYouAgentController.onPageLoad(NormalMode, userAnswers.draftId)
+      case Some(OrganisationAdmin)     =>
+        WhoAreYouAgentController.onPageLoad(NormalMode, userAnswers.draftId)
+      case Some(OrganisationAssistant) =>
+        WhoAreYouAgentController.onPageLoad(NormalMode, userAnswers.draftId)
+      case Some(Agent)                 =>
+        WhoAreYouAgentController.onPageLoad(NormalMode, userAnswers.draftId)
+      case _                           =>
         UnauthorisedController.onPageLoad
     }
```
Comment (RawToast) [inline app/navigation/Navigator.scala:103]: This was the only part with a conflict when rebasing. I took the original PRs changes and add the line for `Some(Agent)` as I believe that would've been the intention had the type existed

**Comment by hmrc-web-operations** [comment]
Thank you for submitting a pull request on this HMRC repository. A member of the HMRC team will review your changes and action accordingly.

HMRC reviewer: please reference [this runbook](https://github.com/hmrc/build-and-deploy/blob/main/runbooks/reviewing_external_pull_requests.md) for guidance on how to proceed.

File: app/models/WhoAreYouAgent.scala (hunk 4 above)
Comment (janosgyenge-digital) [reply app/models/WhoAreYouAgent.scala]: Removed the extra copyright block

File: app/controllers/WhoAreYouAgentController.scala (hunk 3 above)
Comment (janosgyenge-digital) [reply app/controllers/WhoAreYouAgentController.scala]: Refactored it

File: app/controllers/AccountHomeController.scala [hunk 7]
```diff
@@ -34,7 +34,7 @@ import controllers.routes.UnauthorisedController
 import models.{ApplicationForAccountHome, AuthUserType, CounterId, DraftId, NormalMode, UserAnswers}
 import models.requests.DraftSummary
 import navigation.Navigator
-import pages.AccountHomePage
+import pages.{AccountHomePage, WhoAreYouAgentPage}
```
[the remark this answers is not in the document]
Comment (janosgyenge-digital) [reply app/controllers/AccountHomeController.scala]: Removed the unnecessary import

File: test/navigation/NavigatorSpec.scala (hunk 5 above)
Comment (janosgyenge-digital) [reply test/navigation/NavigatorSpec.scala]: Removed

**Comment by hmrc-web-operations** [comment]
Thank you for submitting a pull request on this HMRC repository. A member of the HMRC team will review your changes and action accordingly.

HMRC reviewer: please reference [this runbook](https://github.com/hmrc/build-and-deploy/blob/main/runbooks/reviewing_external_pull_requests.md) for guidance on how to proceed.

File: conf/application.conf [hunk 8]
```diff
@@ -181,6 +181,7 @@ tracking-consent-frontend {
 
 features {
   welsh-translation: true
+  traderDetailsCacheEnabled: false
```
Comment (janosgyenge-digital) [inline conf/application.conf:184]: This flag was copied verbatim from another pull request, so the name is totally misleading. I'll change it to something that better describes the feature.

**Comment by hmrc-web-operations** [comment]
Thank you for submitting a pull request on this HMRC repository. A member of the HMRC team will review your changes and action accordingly.

HMRC reviewer: please reference [this runbook](https://github.com/hmrc/build-and-deploy/blob/main/runbooks/reviewing_external_pull_requests.md) for guidance on how to proceed.

**Comment by hmrc-web-operations** [comment]
Thank you for submitting a pull request on this HMRC repository. A member of the HMRC team will review your changes and action accordingly.

HMRC reviewer: please reference [this runbook](https://github.com/hmrc/build-and-deploy/blob/main/runbooks/reviewing_external_pull_requests.md) for guidance on how to proceed.

File: conf/application.conf (hunk 8 above)
Comment (janosgyenge-digital) [reply conf/application.conf]: Done

**Review by RobertLaverick — APPROVED** [review]
Looks good

File: app/navigation/Navigator.scala (hunk 6 above)
Comment (janosgyenge-digital) [reply app/navigation/Navigator.scala:103]: I can see that this area needs a bit of a cleanup. I'm keen to merge this current PR as is, and raise a technical task to standardise how we display content when the same page looks different depending on the user journey (which may have happened already).

**Pull request merged**